ui/src/app/kvm/components/KVMConfigurationCard/KVMConfigurationCard.test.tsx

```diff
@@ -1,126 +1,202 @@
+    }),
+    resourcepool: resourcePoolStateFactory({
+      items: [resourcePoolFactory({ id: 2 })],
+      loaded: true,
+    }),
+    zone: zoneStateFactory({
+      items: [zoneFactory({ id: 3 })],
+      loaded: true,
+    }),
+  });
+});
 
-  beforeEach(() => {
-    state = rootStateFactory({
-      pod: podStateFactory({
-        items: [podFactory({ id: 1, name: "pod1" })],
-        loaded: true,
-      }),
-    });
+it("can handle updating a lxd KVM", async () => {
+  const pod = podFactory({
+    id: 1,
+    tags: ["tag1", "tag2"],
+    type: PodType.LXD,
   });
+  const store = mockStore(state);
+  render(
+    <Provider store={store}>
+      <MemoryRouter
+        initialEntries={[{ pathname: "/kvm/1/edit", key: "testKey" }]}
+      >
+        <KVMConfigurationCard pod={pod} />
+      </MemoryRouter>
+    </Provider>
+  );
 
-  it("can handle updating a lxd KVM", () => {
-    const pod = podFactory({ id: 1, type: PodType.LXD });
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/kvm/1/edit", key: "testKey" }]}
-        >
-          <KVMConfigurationCard pod={pod} />
-        </MemoryRouter>
-      </Provider>
+  await waitFor(() => {
+    fireEvent.change(screen.getByRole("combobox", { name: "Zone" }), {
+      target: { value: "3" },
+    });
+    fireEvent.change(screen.getByRole("combobox", { name: "Resource pool" }), {
+      target: { value: "2" },
+    });
+    fireEvent.change(screen.getByRole("slider", { name: "CPU overcommit" }), {
+      target: { value: "5" },
+    });
+    fireEvent.change(
+      screen.getByRole("slider", { name: "Memory overcommit" }),
+      { target: { value: "7" } }
     );
+  });
+  await waitFor(() => {
+    fireEvent.click(screen.getByRole("button", { name: "Save changes" }));
+  });
 
-    act(() =>
-      submitFormikForm(wrapper, {
-        cpu_over_commit_ratio: 2,
-        memory_over_commit_ratio: 2,
-        pool: "1",
-        power_address: "192.168.1.1",
-        tags: ["tag1", "tag2"],
-        type: PodType.LXD,
-        zone: "2",
-      })
+  const expectedAction = podActions.update({
+    cpu_over_commit_ratio: 5,
+    id: pod.id,
+    memory_over_commit_ratio: 7,
+    pool: 2,
+    power_address: pod.power_parameters.power_address,
+    tags: "tag1,tag2",
+    type: PodType.LXD,
+    zone: 3,
+  });
+  expect(
+    store.getActions().find((action) => action.type === expectedAction.type)
+  ).toStrictEqual(expectedAction);
+});
+
+it("can handle updating a virsh KVM", async () => {
+  const pod = podFactory({
+    id: 1,
+    tags: ["tag1", "tag2"],
+    type: PodType.VIRSH,
+  });
+  const store = mockStore(state);
+  render(
+    <Provider store={store}>
+      <MemoryRouter
+        initialEntries={[{ pathname: "/kvm/1/edit", key: "testKey" }]}
+      >
+        <KVMConfigurationCard pod={pod} />
+      </MemoryRouter>
+    </Provider>
+  );
+
+  await waitFor(() => {
```

Comment:
These `waitFor` seem arbitrary. It's usually be better to be explicit (and avoid focusing implementation details) as much as possible.

If you're waiting for the button to be enabled after interacting with the form inputs, it might be better to use something along the lines of:

```jsx
await waitFor(() => expect(screen.getByRole("button", { name: "Save changes" }).toBeEnabled())
```

Will also make this test easier to read and help other devs understand the expected behaviour.

Reply:
Good point, thanks!

Reply:
Opened a PR to add eslint-plugin-testing-library https://github.com/canonical-web-and-design/maas-ui/pull/3644 which will give a warning for such a use of `waitFor`